language/stdlib/modules/Roles.move

```diff
@@ -0,0 +1,282 @@
+address 0x1 {
+module Roles {
+    use 0x1::Signer;
+    use 0x1::CoreAddresses;
+
+    /// The roleId contains both the role id and the account address that
+    /// it was created for. This way when privileges are granted to the
+    /// role, we have a way of linking the passed in `account: &signer`
+    /// and `role_id: &RoleId` together in a module-local manner.
+    resource struct RoleId {
+        role_id: u64,
+        owner_address: address,
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Privileges
+    ///////////////////////////////////////////////////////////////////////////
+
+    /// Privileges are extracted in to capabilities. Capabilities hold both
+    /// the privilege witness, along with the account address that they
+    /// were extracted from (i.e. tagged or "tainted"). Capabilities can then only be restored to the
+    /// account where they were extracted from.
+    resource struct Capability<Privilege: resource> {
+        owner_address: address,
+        witness: Privilege,
+    }
+
+    /// The internal representation of of a privilege. We wrap every
+    /// privilege witness resource here to avoid having to write extractors/restorers
+    /// for each privilege, but can instead write this generically.
+    resource struct Privilege<Priv: resource>  {
+        witness: Priv
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Privilege Witnesses
+    ///////////////////////////////////////////////////////////////////////////
+
+    /// The set of possible privileges that we have.
+    resource struct RegisterNewCurrency {}
+    resource struct UpdateDualAttestationThreshold {}
+    resource struct AccountFreezing {}
+    resource struct AccountUnfreezing {}
+    resource struct UpdateVASPCredentials {}
+    resource struct SendWriteSetTransaction {}
+    resource struct UpdateLibraProtocolVersion {}
+    resource struct UpdateGasCosts {}
+    resource struct UpdateTransactionScriptAllowedList {}
+    resource struct PublishModule {}
+    resource struct CreateOnChainConfig {}
+    resource struct RotateComplianceKey {}
+    resource struct UpdateValidatorConfig {}
+
+    // New privileges, need to discuss.
+    resource struct CreatePreburn {}
+    resource struct CreateParentVASP {}
+    resource struct CreateDesignatedDealer {}
+    resource struct CreateTreasuryCompliance {}
+    resource struct CreateValidator {}
+    resource struct CreateSlidingNonce {}
+    resource struct TriggerReconfiguration {}
+    resource struct CreateChildVASP {}
+
+    resource struct AddTier {}
+    resource struct UpdateTier {}
+
+    /*
+    // QUESTION: Do we want these here? Or should they be defined in `Libra`? I'm leaning towards `Libra`.
```

Comment:
I feel it would be good to have all role and capability related stuff somewhere centralized.